src/coreclr/jit/morph.cpp

```diff
@@ -9675,7 +9675,25 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
                 // We do not need to throw if the second operand is a non-(negative one) constant.
                 if (!op2->IsIntegralConst() || op2->IsIntegralConst(-1))
                 {
-                    fgAddCodeRef(compCurBB, bbThrowIndex(compCurBB), SCK_OVERFLOW);
+                    bool checkDividend = true;
+
+                    // We do not need to throw if we know the first operand is a constant and not a min-value.
+                    if (op1->IsIntegralConst())
```

Comment:
Consider using `IsNeverNegative` here. E.g. as I stated in https://github.com/dotnet/runtime/issues/46010 you also don't need overflow check for `GT_ARR_LEN div X`

Reply:
Gotcha - I will check for that then.